--- /home/yuntong/vulnfix/data/libxml2/cve_2017_5969/source/fix-file-backup.c	2022-12-08 03:44:01.644875614 +0000
+++ /home/yuntong/vulnfix/data/libxml2/cve_2017_5969/source/valid.c	2022-12-08 09:53:44.925533977 +0000
@@ -1178,9 +1178,9 @@
 	    else
 		xmlDumpElementContent(buf, content->c1, 0);
             xmlBufferWriteChar(buf, " , ");
-	    if ((content->c2->type == XML_ELEMENT_CONTENT_OR) ||
+	    if (content->c2 != NULL && ((content->c2->type == XML_ELEMENT_CONTENT_OR) ||
 	        ((content->c2->type == XML_ELEMENT_CONTENT_SEQ) &&
-		 (content->c2->ocur != XML_ELEMENT_CONTENT_ONCE)))
+		 (content->c2->ocur != XML_ELEMENT_CONTENT_ONCE))))
 		xmlDumpElementContent(buf, content->c2, 1);
 	    else
 		xmlDumpElementContent(buf, content->c2, 0);
