--- /home/yuntong/vulnfix/data/libtiff/cve_2017_7601/source/libtiff/fix-file-backup.c	2022-12-08 03:41:40.869928396 +0000
+++ /home/yuntong/vulnfix/data/libtiff/cve_2017_7601/source/libtiff/tif_jpeg.c	2022-12-08 09:08:41.574383686 +0000
@@ -1638,6 +1638,7 @@
 		 * default value is inappropriate for YCbCr.  Fill in the
 		 * proper value if application didn't set it.
 		 */
+    if (!(td->td_bitspersample <= 62)) exit(1);
 		{
 			float *ref;
 			if (!TIFFGetField(tif, TIFFTAG_REFERENCEBLACKWHITE,
